C/services/core/configuration_manager.cpp

```diff
@@ -56,53 +57,79 @@ ConfigurationManager* ConfigurationManager::getInstance(const string& host,
  *
  * @return	ConfigCategories class object with
  *		key and description for all found categories.
- * @throw	CategoryDetailsException exception
+ * @throw	CategoryDetailsEx exception
  */
 ConfigCategories ConfigurationManager::getAllCategoryNames() const
 {
+	// Return object
+	ConfigCategories categories;
+
 	vector<Returns *> columns;
 	columns.push_back(new Returns("key"));
 	columns.push_back(new Returns("description"));
 	Query qAllCategories(columns);
 
-	// Query via Storage client
-	ResultSet* allCategories = m_storage->queryTable("configuration", qAllCategories);
-	if (!allCategories)
+	ResultSet* allCategories = 0;
+	try
 	{
-		throw StorageOperationException();
-	}
+		// Query via Storage client
+		allCategories = m_storage->queryTable("configuration", qAllCategories);
+		if (!allCategories)
+		{
+			// Data layer error
+			throw CategoryDetailsEx();
+		}
+		if (!allCategories->rowCount())
```

Comment:
You could combine you tests if you wanted to make the code more concise

Reply:
yes, done